services/gundeck/src/Gundeck/API/Internal.hs

```diff
@@ -16,76 +16,105 @@
+    servantSitemap,
   )
 where
 
 import Cassandra qualified
 import Control.Lens (view)
+import Data.Aeson (eitherDecode)
+import Data.CommaSeparatedList
 import Data.Id
+import Data.Typeable
 import Gundeck.Client qualified as Client
 import Gundeck.Monad
 import Gundeck.Presence qualified as Presence
 import Gundeck.Push qualified as Push
 import Gundeck.Push.Data qualified as PushTok
 import Gundeck.Push.Native.Types qualified as PushTok
-import Imports hiding (head)
-import Network.Wai
-import Network.Wai.Predicate hiding (setStatus)
-import Network.Wai.Routing hiding (route)
-import Network.Wai.Utilities
+import Gundeck.Types.Presence as GD
+import Gundeck.Types.Push.V2
+import Imports
+import Network.Wai (lazyRequestBody)
+import Servant
+import Servant.Server.Internal.Delayed
+import Servant.Server.Internal.DelayedIO
+import Servant.Server.Internal.ErrorFormatter
 import Wire.API.Push.Token qualified as PushTok
-
-sitemap :: Routes a Gundeck ()
-sitemap = do
-  head "/i/status" (continue $ const (pure empty)) true
-  get "/i/status" (continue $ const (pure empty)) true
-
-  -- Push API -----------------------------------------------------------
-
-  post "/i/push/v2" (continue pushH) $
-    request .&. accept "application" "json"
-
-  -- Presence API ----------------------------------------------------------
-
-  get "/i/presences/:uid" (continue Presence.list) $
-    param "uid" .&. accept "application" "json"
-
-  get "/i/presences" (continue Presence.listAll) $
-    param "ids" .&. accept "application" "json"
-
-  post "/i/presences" (continue Presence.add) $
-    request .&. accept "application" "json"
-
-  delete "/i/presences/:uid/devices/:did/cannons/:cannon" (continue Presence.remove) $
-    param "uid" .&. param "did" .&. param "cannon"
-
-  -- User-Client API -------------------------------------------------------
-
-  delete "/i/clients/:cid" (continue unregisterClientH) $
-    header "Z-User" .&. param "cid"
-
-  delete "/i/user" (continue removeUserH) $
-    header "Z-User"
-
-  get "/i/push-tokens/:uid" (continue getPushTokensH) $
-    param "uid"
-
-type JSON = Media "application" "json"
-
-pushH :: Request ::: JSON -> Gundeck Response
-pushH (req ::: _) = do
-  ps <- fromJsonBody (JsonRequest req)
-  empty <$ Push.push ps
-
-unregisterClientH :: UserId ::: ClientId -> Gundeck Response
-unregisterClientH (uid ::: cid) = empty <$ Client.unregister uid cid
-
-removeUserH :: UserId -> Gundeck Response
-removeUserH uid = empty <$ Client.removeUser uid
-
-getPushTokensH :: UserId -> Gundeck Response
-getPushTokensH = fmap json . getPushTokens
-
-getPushTokens :: UserId -> Gundeck PushTok.PushTokenList
-getPushTokens uid = PushTok.PushTokenList <$> (view PushTok.addrPushToken <$$> PushTok.lookup uid Cassandra.All)
+import Wire.API.Routes.Public
+
+-- | this can be replaced by `ReqBody '[JSON] Presence` once the fix in cannon from
+-- https://github.com/wireapp/wire-server/pull/4246 has been deployed everywhere.
```

Comment:
It would maybe be good to mention the purpose of the hack (i.e. making the content type header optional).